src/components/songs/Songs.js

```diff
@@ -0,0 +1,119 @@
+import React, {Component} from 'react';
+import {withStyles} from '@material-ui/core/styles';
+import {connect} from 'react-redux';
+import {FindSongFetch, SongsFetch} from '../../store/songs/actions'
+import Container from "@material-ui/core/Container";
+import Song from './Song'
+import SearchForm from "./SearchForm";
+
+const styles = theme => ({
+    container: {
+        marginTop: theme.spacing(2),
+        flexDirection: 'column',
+        alignItems: 'center',
+    }
+});
+//https://www.pluralsight.com/guides/how-to-implement-infinite-scrolling-with-reactjs
+
+class Songs extends Component {
+    state = {
+        songs: [],
+        loading: false,
+        page: 1,
+        prevY: 0,
+        limit: 2
+    };
+
+    getSongs(page) {
+        const {songsState, SongsFetch} = this.props;
+        this.setState({loading: true});
+        if (songsState.songs.length !== 0) {
+            SongsFetch(page, this.state.limit);
+        }
+        this.setState({loading: false});
+    }
+
+    handleObserver(entities) {
+        const y = entities[0].boundingClientRect.y;
+        if (this.state.prevY > y) {
+            const curPage = this.state.page + 1
+            this.getSongs(curPage);
+            this.setState({page: curPage});
+        }
+        this.setState({prevY: y});
+    }
+
+    componentDidMount() {
+        const {SongsFetch} = this.props;
+
+            SongsFetch(this.state.page, this.state.limit);
+            const options = {
+                root: null,
+                rootMargin: "0px",
+                threshold: 1.0
+            };
+
+            this.observer = new IntersectionObserver(
+                this.handleObserver.bind(this),
```

Comment:
you can use arrow function for handleObserver and avoid creating new function here using ind method